storage/innobase/handler/ha_innodb.cc

```diff
@@ -15741,6 +15741,22 @@ ha_innobase::extra(
 		break;
 	case HA_EXTRA_END_ALTER_COPY:
 		trx = check_trx_exists(ha_thd());
+		if (m_prebuilt->table->skip_alter_undo) {
+			if (dberr_t err= trx->bulk_insert_apply()) {
+				if (err == DB_DUPLICATE_KEY) {
+					return HA_ERR_FOUND_DUPP_KEY;
+				}
+				return 1;
+			}
+
+			trx->end_bulk_insert(*m_prebuilt->table);
+			trx->bulk_insert = false;
+			if (!dict_stats_is_persistent_enabled(
+					m_prebuilt->table)) {
+				dict_stats_update_if_needed(
+					m_prebuilt->table, *trx);
+			}
```

Comment:
It would be nice to have a comment on why this is needed only if `STATS_PERSISTENT=0`.